docs/02-app/02-api-reference/02-file-conventions/middleware.mdx

```diff
@@ -0,0 +1,135 @@
+---
+title: middleware.js
+description: API reference for the middleware.js  file.
+related:
+  title: Learn more about Middleware
+  links:
+    - app/building-your-application/routing/middleware
+---
+
+The `middleware.js|ts` file is used to write [Middleware](/docs/app/building-your-application/routing/middleware) and run code on the server before a request is completed. Then, based on the incoming request, you can modify the response by rewriting, redirecting, modifying the request or response headers, or responding directly.
+
+Middleware executes before routes are rendered. It's particularly useful for implementing custom server-side logic like authentication, logging, or handling redirects.
+
+Use the file `middleware.ts` (or .js) in the root of your project to define Middleware. For example, at the same level as `app` or `pages`, or inside `src` if applicable.
+
+```tsx filename="middleware.ts" switcher
+import { NextResponse, NextRequest } from 'next/server'
+
+// This function can be marked `async` if using `await` inside
+export function middleware(request: NextRequest) {
+  return NextResponse.redirect(new URL('/home', request.url))
+}
+
+export const config = {
+  matcher: '/about/:path*',
+}
+```
+
+```js filename="middleware.js" switcher
+import { NextResponse } from 'next/server'
+
+// This function can be marked `async` if using `await` inside
+export function middleware(request) {
+  return NextResponse.redirect(new URL('/home', request.url))
+}
+
+export const config = {
+  matcher: '/about/:path*',
+}
+```
+
+## Exports
+
+### Middleware function
+
+The file must export a single function, either as a default export or named `middleware`. Note that multiple middleware from the same file are not supported.
+
+```js filename="middleware.js"
+// Example of default export
+export default function middleware(request) {
+  // Middleware logic
+}
+```
+
+### Config object (optional)
+
+Optionally, a config object can be exported alongside the Middleware function. This object includes the [matcher](#matcher) to specify paths where the Middleware applies.
+
+#### Matcher
+
+The `matcher` option allows you to target specific paths for the Middleware to run on. You can specify these paths in several ways:
+
+- For a single path: Directly use a string to define the path, like `'/about'`.
+- For multiple paths: Use an array to list multiple paths, such as `matcher: ['/about', '/contact']`, which applies the Middleware to both `/about` and `/contact`.
+
+Additionally, `matcher` supports complex path specifications through regular expressions, such as `matcher: ['/((?!api|_next/static|_next/image|.*\\.png$).*)']`, enabling precise control over which paths to include or exclude.
+
+The `matcher` option also accepts an array of objects with the following keys:
+
+- `source`: The path or pattern used to match the request paths. It can be a simple string for direct path matching or a pattern for more complex matching.
```

Comment:
```suggestion
- `source`: The path or pattern used to match the request paths. It can be a string for direct path matching or a pattern for more complex matching.
```